src/main/java/io/logz/jmx2logzio/listener/prometheus/PromLogzioSender.kt

```diff
@@ -0,0 +1,128 @@
+package io.logz.jmx2logzio.listener.prometheus
+
+import io.logz.prometheus.protobuf.Types
+import io.logz.sender.HttpsRequestConfiguration
+import io.logz.sender.SenderStatusReporter
+import io.logz.sender.exceptions.LogzioServerErrorException
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+import java.util.concurrent.atomic.AtomicBoolean
+
+class PromLogzioSender(
+    private val reporter: SenderStatusReporter,
+    private val tasksExecutor: ScheduledExecutorService,
+    private val timeSeriesQueue: TimeSeriesDiskQueue,
+    private val debug: Boolean = false,
+    private val drainTimeoutSec: Long = 5,
+    httpsRequestConfiguration: HttpsRequestConfiguration
+) {
+    companion object {
+        private const val MAX_SIZE_IN_BYTES = 3 * 1024 * 1024 // 3 MB
+        private const val FINAL_DRAIN_TIMEOUT_SEC = 20
+    }
+
+    private val promHttpsSyncSender = PromHttpsSyncSender(httpsRequestConfiguration, reporter)
+    private val drainRunning = AtomicBoolean(false)
+
+    init {
+        debug("Created new LogzioPrometheusSender class")
+    }
+
+    fun start() {
+        tasksExecutor.scheduleAtFixedRate(this::drainQueueAndSend, 0, drainTimeoutSec, TimeUnit.SECONDS)
```

Comment:
Actually that can create a Queue which is not really needed. Replace Rate with Interval - this is the proper method.